src/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -750,6 +763,36 @@ private static void CheckBaseAddress(Uri baseAddress, string parameterName)
 
         private HttpRequestMessage CreateRequestMessage(HttpMethod method, Uri uri) =>
             new HttpRequestMessage(method, uri) { Version = _defaultRequestVersion };
+
+        private Exception CreateTimeoutException(Exception innerException)
+        {
+            return new HttpRequestException(
+                SR.net_http_timeout,
+                new TimeoutException(SR.net_http_timeout, innerException),
+                true);
```

Comment:
I wasn't sure about this, to be honest. I thought it would make sense to `allowRetry` for a timeout, but I don't really know how it's used internally. I'll change it to false.